### PR TITLE
Split gofiles and add generator in download-geofabrik

### PR DESCRIPTION
**Pull request by julien-noblet**
* Indroducing the ```generate``` command
* Removing old generator.go
* Add better verbose and quiet options

### CODE REVIEW

File: generator_test.go [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+		myConfig *Config
+	}
+	tests := []struct {
+		name    string
+		e       ElementSlice
+		args    args
+		want    []byte
+		wantErr bool
+	}{
+		// TODO: Add test cases.
```
Comment (julien-noblet) [inline generator_test.go]: File is not gofmt-ed with -s

Comment (julien-noblet) [inline generator_test.go]: File is not gofmt-ed with -s

Comment (julien-noblet) [inline generator_test.go]: File is not gofmt-ed with -s

Comment (julien-noblet) [inline generator_test.go]: File is not gofmt-ed with -s

Comment (julien-noblet) [inline generator_test.go]: File is not gofmt-ed with -s

File: download-geofabrik_test.go [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+	}
+	for _, tt := range tests {
+		*fService = tt.service
+		t.Run(tt.name, func(t *testing.T) {
+			if tt.config != "" {
+				*fConfig = tt.config
+			}
+			if got := checkService(); got != tt.want {
+				t.Errorf("checkService() = %v, want %v", got, tt.want)
+			}
```
Comment (julien-noblet) [inline download-geofabrik_test.go]: Error return value is not checked

File: download-geofabrik.go [hunk 3]
```diff
@@ -1,56 +1,33 @@
...
-type config struct {
-	BaseURL  string             `yaml:"baseURL"`
-	Formats  map[string]format  `yaml:"formats,flow"`
-	Elements map[string]element `yaml:"elements,flow"`
-}
-
-type element struct {
-	ID     string   `yaml:"id"`
-	File   string   `yaml:"file"`
-	Meta   bool     `yaml:"meta"`
```
Comment (julien-noblet) [inline download-geofabrik.go]: Error return value is not checked

File: download-geofabrik.go [hunk 4]
```diff
@@ -1,56 +1,33 @@
...
-	ID  string `yaml:"ext"`
-	Loc string `yaml:"loc"`
-}
-
 var (
 	app         = kingpin.New("download-geofabrik", "A command-line tool for downloading OSM files.")
+	fService    = app.Flag("service", "Can switch to another service. You can use \"geofabrik\" or \"openstreetmap.fr\". It automatically change config file if -c is unused.").Default("geofabrik").String()
 	fConfig     = app.Flag("config", "Set Config file.").Default("./geofabrik.yml").Short('c').String()
-	nodownload  = app.Flag("nodownload", "Do not download file (test only)").Short('n').Bool()
-	verbose     = app.Flag("verbose", "Be verbose").Short('v').Bool()
```
Comment (julien-noblet) [inline download-geofabrik.go]: Error return value is not checked

File: download-geofabrik_test.go [hunk 5]
```diff
@@ -0,0 +1,184 @@
...
+		// TODO: Add test cases.
+		{name: "checkService(), fService = geofabrik", service: "geofabrik", want: true},
+		{name: "checkService(), fService = openstreetmap.fr", service: "openstreetmap.fr", config: "./geofabrik.yml", want: true, wantConfig: "./openstreetmap.fr.yml"},
+		{name: "checkService(), fService = anothermap", service: "anothermap", want: false},
+		{name: "checkService(), fService = \"\"", service: "", want: false},
+	}
+	for _, tt := range tests {
+		*fService = tt.service
+		t.Run(tt.name, func(t *testing.T) {
+			if tt.config != "" {
```
Comment (julien-noblet) [inline download-geofabrik_test.go]: Error return value is not checked

File: download-geofabrik.go (hunk 3 above)
Comment (julien-noblet) [inline download-geofabrik.go]: Error return value is not checked

File: download-geofabrik.go (hunk 4 above)
Comment (julien-noblet) [inline download-geofabrik.go]: Error return value is not checked

File: download-geofabrik_test.go [hunk 6]
```diff
@@ -0,0 +1,200 @@
...
```
Comment (julien-noblet) [inline download-geofabrik_test.go]: Error return value is not checked

File: download-geofabrik.go [hunk 7]
```diff
@@ -1,56 +1,33 @@
...
-	"golang.org/x/net/proxy"
-
 	"github.com/olekukonko/tablewriter"
-
 	"gopkg.in/alecthomas/kingpin.v2"
-	"gopkg.in/yaml.v2"
 )
 
-type config struct {
-	BaseURL  string             `yaml:"baseURL"`
```
Comment (julien-noblet) [inline download-geofabrik.go]: Error return value is not checked

File: download-geofabrik_test.go [hunk 8]
```diff
@@ -0,0 +1,200 @@
...
+				*fConfig = tt.config
+			}
```
Comment (julien-noblet) [inline download-geofabrik_test.go]: Error return value is not checked

File: download-geofabrik.go (hunk 7 above)
Comment (julien-noblet) [reply download-geofabrik.go]: ???

File: download-geofabrik_test.go (hunk 8 above)
Comment (julien-noblet) [reply download-geofabrik_test.go]: I will no check for error from t.Errorf....

File: download-geofabrik.go (hunk 7 above)
Comment (julien-noblet) [inline download-geofabrik.go]: Error return value is not checked

File: download-geofabrik_test.go [hunk 9]
```diff
@@ -0,0 +1,201 @@
+package main
+
+import (
+	"fmt"
```
Comment (julien-noblet) [inline download-geofabrik_test.go]: Error return value is not checked

Comment (julien-noblet) [inline download-geofabrik_test.go]: Error return value is not checked

Comment (julien-noblet) [inline download-geofabrik_test.go]: Error return value is not checked

Comment (julien-noblet) [inline download-geofabrik_test.go]: Error return value is not checked

**Comment by ImreSamu** [comment]
Let go !  :)

File: download-geofabrik_test.go (hunk 9 above)
Comment (julien-noblet) [inline download-geofabrik_test.go]: Error return value is not checked

Comment (julien-noblet) [inline download-geofabrik_test.go]: Error return value is not checked

**Comment by julien-noblet** [comment]
Done :smile:

**Pull request merged**